processing/src/main/java/io/druid/query/metadata/SegmentMetadataQueryRunnerFactory.java

```diff
@@ -87,7 +87,8 @@ public SegmentMetadataQueryRunnerFactory(
       public Sequence<SegmentAnalysis> run(QueryPlus<SegmentAnalysis> inQ, Map<String, Object> responseContext)
       {
         SegmentMetadataQuery query = (SegmentMetadataQuery) inQ.getQuery();
-        final SegmentAnalyzer analyzer = new SegmentAnalyzer(query.getAnalysisTypes());
+        SegmentMetadataQuery updatedQuery = query.withFinalizedAnalysisTypes(toolChest.getConfig());
```

Comment:
Might as well do this and the cast in one line to avoid accidentally referencing `query` instead of `updatedQuery`.